### PR TITLE
Separate non-DUNE methods in LevelCartesianIndexMapper<AluGrid>

### PR DESCRIPTION
**Pull request by aritorto**
Needs OPM/opm-grid#766

### CODE REVIEW

File: opm/simulators/flow/AluGridVanguard.hpp [hunk 1]
```diff
@@ -232,6 +235,13 @@ class AluGridVanguard : public FlowBaseVanguard<TypeTag>
     const CartesianIndexMapper& cartesianIndexMapper() const
     { return *cartesianIndexMapper_; }
 
+    /*!
+     * \brief Returns the object which maps a global element index of the simulation grid
+     *        to the corresponding element index of the level logically Cartesian index.
+     */
+    const LevelCartesianIndexMapper levelCartesianIndexMapper() const // Consider adding data member to reduce/optimize memory-use?
```
Comment (lisajulia) [inline opm/simulators/flow/AluGridVanguard.hpp:242]: Resolve the comment please.

File: opm/simulators/flow/AluGridVanguard.hpp [hunk 2]
```diff
@@ -34,10 +34,12 @@
 #include <opm/common/OpmLog/OpmLog.hpp>
 
 #include <opm/grid/CpGrid.hpp>
+#include <opm/grid/cpgrid/LevelCartesianIndexMapper.hh>
```
Comment (lisajulia) [inline opm/simulators/flow/AluGridVanguard.hpp:37]: Also related to https://github.com/OPM/opm-grid/pull/766: why did you choose ".hh" as the extension here and not ".hpp"?

File: opm/simulators/flow/CpGridVanguard.hpp [hunk 3]
```diff
@@ -269,6 +272,13 @@ class CpGridVanguard : public FlowBaseVanguard<TypeTag>
         return this->grid().globalCell();
     }
 
+    /*!
+     * \brief Returns the object which maps a global element index of the simulation grid
+     *        to the corresponding element index of the level logically Cartesian index.
+     */
+    const LevelCartesianIndexMapper levelCartesianIndexMapper() const
```
Comment (lisajulia) [inline opm/simulators/flow/CpGridVanguard.hpp:279]: Is it correct that this function is here and not in GenericCpGridVanguard.hpp? If so, then leave it as it is.

**Review by lisajulia — CHANGES_REQUESTED** [review]
:+1:

File: opm/simulators/flow/AluGridVanguard.hpp (hunk 2 above)
Comment (aritorto) [reply opm/simulators/flow/AluGridVanguard.hpp:37]: I wasn't sure about that. For CpGrid, many (maybe all?) headers have the extension ".hpp". For PolyhedralGird, ".hh", and for AluGrid, ".hpp". Then, should I go for the majority and take ".hpp"?

File: opm/simulators/flow/CpGridVanguard.hpp (hunk 3 above)
Comment (aritorto) [reply opm/simulators/flow/CpGridVanguard.hpp:279]: Oh no! Thanks for catching that. It should be removed.

File: opm/simulators/flow/AluGridVanguard.hpp (hunk 1 above)
Comment (aritorto) [reply opm/simulators/flow/AluGridVanguard.hpp]: Comment has been removed. Decision will be taken in the future with a clearer idea of how many times it needs to be called.

**Comment by aritorto** [comment]
@lisajulia I just solved merge conflict and will run jenkins one more time :)

File: opm/simulators/flow/AluGridVanguard.hpp (hunk 2 above)
Comment (lisajulia) [reply opm/simulators/flow/AluGridVanguard.hpp]: Yes, would say that is the correct approach here,  ".hpp" for ../opm/common/LevelCartesianIndexMapper.hpp, for the CpGrid specialization, and for ALuGrid, and only ".hh" for PolyhedralGrid specialization.

**Review by lisajulia — APPROVED** [review]
Looks fine now :) so you will rewrite the git history now and I can merge then, right?

**Comment by aritorto** [comment]
> Looks fine now :) so you will rewrite the git history now and I can merge then, right?

Great, thanks! I'll do that then!

**Comment by aritorto** [comment]
@lisajulia done! If there are no further comments on OPM/opm-grid#766, I'll run jenkins one last time and they will be ready to be merged!

**Comment by lisajulia** [comment]
jenkins build this please

**Pull request merged**